rllib/utils/nested_dict.py

```diff
@@ -212,7 +210,11 @@ def __iter__(self) -> Iterator[SeqStrType]:
         while stack:
             k, v = stack.pop(0)
             if isinstance(v, NestedDict):
-                stack = [(k + (StrKey(k2),), v) for k2, v in v._data.items()] + stack
+                items = v._data.items()
+                if len(items) == 0:
+                    yield tuple(k)
+                else:
+                    stack = [(k + (StrKey(k2),), v) for k2, v in items] + stack
```

Comment:
Instead of this, can we re-write it as:
```
if len(v) == 0:
     yield tuple(k)
else:
    stack = ...
```
The reason is that computing len(items) is more expensive than len(foo). Because items returns a generator and you have to iterate over the generator to compute its length.

Reply:
We need to add a good explaination in the docstring of the NestedDict to say clarify the iteration behavior. Users should not expect the open roots to show up during iteration and len() computation.

Reply:
Wow nice catch! I'll add the explanation

Reply:
The len(v) actually does not work, since v might be a NestedDict and len(v) for an empty leaf is zero (even though we want to iterate over it). I've changed this to len(v._data), since the length of a dictionary in python is a simple counter and len(v._data) is O(1).

Reply:
sg. Thanks.